lib/Sema/CSGen.cpp

```diff
@@ -2566,8 +2584,15 @@ namespace {
         FunctionRefKind functionRefKind = FunctionRefKind::Compound;
         if (!enumPattern->getParentType().isNull()) {
           // Resolve the parent type.
-          Type parentType =
-            resolveTypeReferenceInExpression(enumPattern->getParentType());
+          Type parentType = resolveTypeReferenceInExpression(
+              enumPattern->getParentType(), [](auto unboundTy) {
```

Comment:
We oughta make a functor for the no-op case a default argument to both functions.

Reply:
The no-op is a temporary hack. I figured writing them out explicitly with FIXMEs is a good idea, rather than defaulting the parameter.